File: 1295. Find Numbers with Even Number of Digits/1295.cpp
#include <iostream>
#include <vector>
using namespace std;

class Solution {
public:
    int findNumbers(vector<int>& nums) {
        int result = 0;
        for (auto &n :nums){
            int tmp = n;
            int digits = 0;
            while (tmp != 0){
                digits++;
                tmp /= 10;
            }
            //cout << digits << endl;
            if (digits%2 == 0){
                result++;
            }
        }
        return result;
    }
};

int main(){
    Solution s;
    vector<int> nums = {12,345,2,6,7896};
    cout << s.findNumbers(nums) << endl;
    return 0;
}

// compile: g++ -std=c++11 1295.cpp -o 1295
// execute: ./1295

File: 1295. Find Numbers with Even Number of Digits/1295.py
class Solution(object):
    def findNumbers(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        result = 0
        for n in nums:
            digits = 0
            while (n != 0):
                digits += 1
                n /= 10
            if (digits%2 == 0):
                result += 1
        return result
    
if __name__ == '__main__':
    s = Solution()
    nums = [12,345,2,6,7896]
    print(s.findNumbers(nums))

# python3 1295.py